### PR TITLE
[cli] add `netstat` command

### PR DESCRIPTION
**Pull request by Moandor-y**
This fixes https://github.com/openthread/openthread/issues/5186 by adding `netstat` command to CLI.

### CODE REVIEW

File: src/cli/cli.cpp [hunk 1]
```diff
@@ -2062,6 +2065,66 @@ void Interpreter::ProcessNetif(uint8_t aArgsLength, char *aArgs[])
 }
 #endif
 
+#if OPENTHREAD_CONFIG_PLATFORM_UDP_ENABLE
```
Comment (jwhui) [inline src/cli/cli.cpp:2068]: What's the motivation for using `OPENTHREAD_CONFIG_PLATFORM_UDP_ENABLE` feature flag?

File: src/cli/cli.hpp [hunk 2]
```diff
@@ -177,8 +177,12 @@ class Interpreter
      * Write an IPv6 address to the CLI console.
      *
      * @param[in]  aAddress  A reference to the IPv6 address.
+     *
+     * @returns The number of bytes placed in the output queue.
+     *
+     * @retval  -1  Driver is broken.
```
Comment (jwhui) [inline src/cli/cli.hpp:183]: minor nit to be consistent:
```suggestion
     * @retval  -1  Driver is broken.
     *
```

File: src/cli/cli.cpp [hunk 3]
```diff
@@ -2062,6 +2065,66 @@ void Interpreter::ProcessNetif(uint8_t aArgsLength, char *aArgs[])
...
+    }
+
+    AppendResult(OT_ERROR_NONE);
+}
+
+int Interpreter::OutputSocketAddress(const otSockAddr &aAddress)
+{
+    int outputLength = 0;
+
+    outputLength += OutputIp6Address(aAddress.mAddress);
```
Comment (jwhui) [inline src/cli/cli.cpp:2112]: Check negative return value here and below?

File: src/cli/cli.cpp (hunk 1 above)
Comment (Moandor-y) [reply src/cli/cli.cpp:2068]: `otUdpGetSockets` is defined only if `OPENTHREAD_CONFIG_PLATFORM_UDP_ENABLE=1`.

Comment (jwhui) [reply src/cli/cli.cpp:2068]: Yes, but why? 😄 @bukepo , thoughts?

Comment (abtink) [reply src/cli/cli.cpp:2068]: I noticed the same in `udp_api.cpp` (though it does not explicitly mention this in the `udp.h` documentation of this API). I guess when it was added it was only used/intended when `PLATFORM_UDP` feature was enabled. 

I also favor that we do not restrict this API to this config and make available in general.

File: src/cli/cli.cpp (hunk 3 above)
Comment (Moandor-y) [reply src/cli/cli.cpp]: Done.

File: src/cli/cli.cpp (hunk 1 above)
Comment (bukepo) [reply src/cli/cli.cpp:2068]: Abtin helped answer this question, and yes, agree not to restrict this API.

Comment (Moandor-y) [reply src/cli/cli.cpp]: Removed this flag requirement from API/CLI.

File: src/core/net/udp6.cpp [hunk 4]
```diff
@@ -124,12 +124,19 @@ otError Udp::Socket::Connect(const SockAddr &aSockAddr)
 
     mPeerName = aSockAddr;
 
+    if (!IsBound())
+    {
+        SuccessOrExit(error = Bind(GetSockName()));
+    }
```
Comment (abtink) [inline src/core/net/udp6.cpp:130]: 👍

**Pull request merged**